### ISSUE TITLE
Refactor dependencies

### ISSUE THREAD

**Issue by marioradu05** [dotkernel/dot-session#14]
The autologin (remember me) middleware uses some external dependencies, for example :

```
use Frontend\User\Authentication\PersistentAuthenticationAdapter;
use Frontend\User\Entity\UserIdentity;
use Frontend\User\Entity\UserToken;
use Frontend\User\Service\UserTokenService;
```
Those dependencies should be refactored to use internal interfaces and the external classes will need to implement them.

**Comment by arhimede** [comment]
@MarioRadu decide where to have the rememberMe function 
here or in frontend 

and keep only one

**Comment by arhimede** [comment]
RememberMe function will stay in frontend 

As a standalone feature .

Because it have too many dependencies , too much for current package

**Comment by alexmerlin** [comment]
Closing this issue as it seems to be fixed in #17
